src/seedsigner/views/tools_views.py

```diff
@@ -256,6 +264,194 @@ def run(self):
+        )
+
+        if ret == RET_CODE__BACK_BUTTON:
+            return Destination(BackStackView)
+
+        coin_seed_phrase = mnemonic_generation.generate_mnemonic_from_coin_flips(ret)
+
+        # Add the mnemonic as an in-memory Seed
+        seed = Seed(coin_seed_phrase, wordlist_language_code=self.settings.get_value(SettingsConstants.SETTING__WORDLIST_LANGUAGE))
+        self.controller.storage.set_pending_seed(seed)
+
+        # Cannot return BACK to this View
+        return Destination(SeedWordsWarningView, view_args={"seed_num": None}, clear_history=True)
+
+
+
+class ToolsCoinEntropySetwiseEntryView(View):
+    """ Handles set-wise coin flip entry (11-bit groups for BIP-39 words). """
+    def __init__(self, total_flips: int, current_set: int = 1, bits_collected: str = "", mnemonic: list = None):
+        super().__init__()
+        self.total_flips = total_flips
+        self.current_set = current_set
+        self.bits_collected = bits_collected
+        self.mnemonic = mnemonic if mnemonic is not None else []
+        self.total_sets = 11 if total_flips == 128 else 23
+        self.last_set_bits = 7 if total_flips == 128 else 3
+
+    def run(self):
+        wordlist_language_code = self.settings.get_value(SettingsConstants.SETTING__WORDLIST_LANGUAGE)
+        required_bits = self.last_set_bits if self.current_set == self.total_sets + 1 else 11
+
+        ret = self.run_screen(
+            ToolsCoinEntropySetwiseEntryScreen,
+            current_set=self.current_set,
+            total_sets=self.total_sets + 1,
+            required_bits=required_bits
+        )
+
+        if ret == RET_CODE__BACK_BUTTON:
+            return Destination(ToolsCoinInputMethodView, view_args={"total_flips": self.total_flips})
+
+        self.bits_collected += ret
+
+        if self.current_set <= self.total_sets:
+            # Convert the 11-bit set to a BIP-39 word
+            word_index = int(ret, 2)
+            word = Seed.get_wordlist(wordlist_language_code)[word_index]
+            self.mnemonic.append(word)
+
+        # Navigate to display the word or finalize if done
+        if self.current_set <= self.total_sets:
+            return Destination(
+                ToolsCoinEntropySetwiseBip39WordView,
+                view_args={
+                    "total_flips": self.total_flips,
+                    "current_set": self.current_set,
+                    "bits_collected": self.bits_collected,
+                    "mnemonic": self.mnemonic,
+                    "word": word
+                }
+            )
+        else:
+            # Final set collected; generate the full mnemonic
+            full_entropy = self.bits_collected[:self.total_flips]
+            entropy_bytes = int(full_entropy, 2).to_bytes(16 if self.total_flips == 128 else 32, byteorder='big')
+            mnemonic = mnemonic_generation.generate_mnemonic_from_bytes(entropy_bytes, wordlist_language_code=wordlist_language_code)
```

Comment:
This should be using the `generate_mnemonic_from_coin_flips` utility function.

Even for straightforward operations we have to be able to point to one and only one place that is doing vital calculations that has unit test coverage.